src/lib.rs

```diff
@@ -778,12 +806,12 @@ mod test {
         #[test]
         fn impl_into_iter() {
             let vec = vec1![ 1, 2, 3];
-            assert_eq!(6, vec.into_iter().sum());
+            assert_eq!(6, vec.into_iter().sum::<u8>());
```

Comment:
Bizarrely these type annotations because necessary when the `serde` feature was enabled. Perhaps there's an extra impl of `std::iter::Sum` that confuses the type checker?